test/local.mk

```diff
@@ -101,3 +106,25 @@ $(webhook_key):
 
 $(webhook_cert): $(webhook_key)
 	openssl req -x509 -key $(webhook_key) -nodes -out $@ -days 3650 -subj "/CN=$(webhook_service_name)" -addext "subjectAltName = DNS:$(webhook_service_name)"
+
+###
+### E2E Tests
+### with KUTTL (https://kuttl.dev)
+###
+
+kuttl_bin = $(kind_dir)/kuttl
+
+# Prepare binary
+# We need to set the Go arch since the binary is meant for the user's OS.
+$(kuttl_bin): export GOOS = $(shell go env GOOS)
+$(kuttl_bin): export GOARCH = $(shell go env GOARCH)
+$(kuttl_bin):
+	@mkdir -p $(kind_dir)
+	cd test/e2e && go build -o $@ github.com/kudobuilder/kuttl/cmd/kubectl-kuttl
+
+test-e2e: export KUBECONFIG = $(KIND_KUBECONFIG)
+test-e2e: $(kuttl_bin) local-install provider-config ## E2E tests
+	kubectl create ns --save-config e2e-test --dry-run=client -o yaml | kubectl apply -f -
```

Comment:
Is there a specific reason to create the namespace this way? This seems a bit overcomplicated.

Isn't `kubectl create ns e2e-test` simpler?

Reply:
After a test run, the namespace `e2e-test` remains.
When `kubectl create` is executed a second time, it fails with "namespace exists already" error. Thus I use `kubectl apply` so it's idempotent but I need the YAML manifest first ("--save-config" creates the kubectl annotation)